eslint.config.js

```diff
@@ -0,0 +1,249 @@
+import chaiFriendlyPlugin from 'eslint-plugin-chai-friendly'
+import cypressPlugin from 'eslint-plugin-cypress/flat'
+import jsdocPlugin from 'eslint-plugin-jsdoc'
+import mochaPlugin from 'eslint-plugin-mocha'
+import icedfrisbyPlugin from 'eslint-plugin-icedfrisby'
+import sortClassMembersPlugin from 'eslint-plugin-sort-class-members'
+import importPlugin from 'eslint-plugin-import'
+import reactHooksPlugin from 'eslint-plugin-react-hooks'
+import prettierConfig from 'eslint-plugin-prettier/recommended'
+import promisePlugin from 'eslint-plugin-promise'
+import globals from 'globals'
+import neostandard from 'neostandard'
+import tsParser from '@typescript-eslint/parser'
+import js from '@eslint/js'
+
+// Config that is used across the whole codebase
+// and customisations to built-in ESLint rules
+const globalConfig = {
+  plugins: {
+    import: importPlugin,
+    promise: promisePlugin,
+  },
+
+  rules: {
+    'import/order': ['error', { 'newlines-between': 'never' }],
+    'promise/prefer-await-to-then': 'error',
+
+    // ESLint built-in rules config
+    'no-empty': ['error', { allowEmptyCatch: true }],
+    'no-var': 'error',
+    'prefer-const': 'error',
+    'arrow-body-style': ['error', 'as-needed'],
+    'object-shorthand': ['error', 'properties'],
+    'prefer-template': 'error',
+    'func-style': ['error', 'declaration', { allowArrowFunctions: true }],
+    'new-cap': ['error', { capIsNew: true }],
+    quotes: [
+      'error',
+      'single',
+      { avoidEscape: true, allowTemplateLiterals: false },
+    ],
+    camelcase: [
+      'error',
+      {
+        ignoreDestructuring: true,
+        properties: 'never',
+        ignoreGlobals: true,
+        allow: ['^UNSAFE_'],
+      },
+    ],
+  },
+}
+
+// config specific to linting Node (CommonJS) files
+const commonJsConfig = {
+  files: ['badge-maker/**/*.js', '**/*.cjs'],
+
+  languageOptions: {
+    globals: {
+      ...globals.node,
+    },
+  },
+}
+
+// config specific to linting Node (ESModules) files
+const nodeEsmConfig = {
+  files: ['**/*.@(js|mjs)', '!frontend/**/*.js', '!badge-maker/**/*.js'],
+
+  languageOptions: {
+    globals: {
+      ...globals.node,
+    },
+    parser: tsParser,
+    sourceType: 'module',
+  },
+
+  rules: {
+    'no-console': 'off',
+  },
+}
+
+// config specific to linting Frontend (ESModules) files
+const frontendConfig = {
+  files: ['frontend/**/*.js'],
+
+  plugins: {
+    'react-hooks': reactHooksPlugin,
+  },
+
+  languageOptions: {
+    globals: {
+      ...globals.browser,
+    },
+    sourceType: 'module',
+  },
+
+  rules: {
+    'react-hooks/rules-of-hooks': 'error',
+    'react-hooks/exhaustive-deps': 'error',
+  },
```

Comment:
Now that we have migrated to docusaurus, we don't really have much custom react code any more although there is a little bit.
Looking over our old config, the only rule from eslint-plugin-react we actually had enabled was `react/jsx-sort-props` so I decided to just ditch eslint-plugin-react completely.
I think there is some value in keeping these two rules because mistakes with hooks are common causes of infinite loops